src/app/core/internationalization.module.ts

```diff
@@ -83,12 +81,46 @@ class ICMTranslateLoader implements TranslateLoader {
   }
 }
 
+const FALLBACK_LANG = new InjectionToken<string>('fallbackTranslateLanguage');
+
+@Injectable()
+class FallbackMissingTranslationHandler implements MissingTranslationHandler {
+  constructor(
+    private parser: TranslateParser,
+    private translateLoader: TranslateLoader,
+    private errorHandler: ErrorHandler,
+    @Inject(FALLBACK_LANG) private fallback: string
+  ) {}
+
+  handle(params: MissingTranslationHandlerParams) {
+    if (params.interpolateParams || /^\w+(\.[\w-]+)+$/.test(params.key)) {
+      this.errorHandler.handleError(`missing translation in ${params.translateService.currentLang}: ${params.key}`);
+      if (params.translateService.currentLang !== this.fallback) {
+        return this.translateLoader.getTranslation(this.fallback).pipe(
```

Comment:
While I see how it's nice to consolidate the fallback handling into the MissingTranslationsLoader, the current setup triggers a `getTranslation()` call for each missing translation, resulting in many repetitive calls. It should be possible to cache the fallback translations to avoid this.

Also, the API provides key-specific translation functionality:
![image](https://user-images.githubusercontent.com/34165455/121013526-35b65e80-c799-11eb-806f-0eea0045697e.png)
So in case of a missing translation key (maybe from an error message) it would be nice to check the API for that specific key in addition to querying the fallback translation.

Reply:
✔️

Reply:
I would suggest limiting the individual lookup to the client and only to error messages. Otherwise too many calls would be triggered.

Still haven't got it working perfectly, I'll have another look tomorrow 😄

Reply:
I refactored the feature to use the NgRx store and now it seems to be working 😄